man/mysqld_multi.1
'\" t
.\"
.TH "\fBMARIADBD-MULTI\fR" "1" "15 May 2020" "MariaDB 10.11" "MariaDB Database System"
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.\" mysqld_multi
.\" tools: mysqld_multi
.\" scripts
.\" multi mysqld
.SH "NAME"
mariadbd-multi \- manage multiple MariaDB servers (mysqld_multi is now a symlink to mariadbd-multi)
.SH "SYNOPSIS"
.HP \w'\fBmysqld_multi\ [\fR\fB\fIoptions\fR\fR\fB]\ {start|stop|report}\ [\fR\fB\fIGNR\fR\fR\fB[,\fR\fB\fIGNR\fR\fR\fB]\ \&.\&.\&.]\fR\ 'u
\fBmysqld_multi [\fR\fB\fIoptions\fR\fR\fB] {start|stop|report} [\fR\fB\fIGNR\fR\fR\fB[,\fR\fB\fIGNR\fR\fR\fB] \&.\&.\&.]\fR
.SH "DESCRIPTION"
.PP
\fBmysqld_multi\fR
is designed to manage several
\fBmysqld\fR
processes that listen for connections on different Unix socket files and TCP/IP ports\&. It can start or stop servers, or report their current status\&.
.PP
\fBmysqld_multi\fR
searches for groups named
[mysqld\fIN\fR]
in
my\&.cnf
(or in the file named by the
\fB\-\-config\-file\fR
option)\&.
\fIN\fR
can be any positive integer\&. This number is referred to in the following discussion as the option group number, or
\fIGNR\fR\&. Group numbers distinguish option groups from one another and are used as arguments to
\fBmysqld_multi\fR
to specify which servers you want to start, stop, or obtain a status report for\&. Options listed in these groups are the same that you would use in the
[mysqld]
group used for starting
\fBmysqld\fR\&. However, when using multiple servers, it is necessary that each one use its own value for options such as the Unix socket file and TCP/IP port number\&.
.PP
To invoke
\fBmysqld_multi\fR, use the following syntax:
.sp
.if n \{\
.RS 4
.\}
.nf
shell> \fBmysqld_multi [\fR\fB\fIoptions\fR\fR\fB] {start|stop|report} [\fR\fB\fIGNR\fR\fR\fB[,\fR\fB\fIGNR\fR\fR\fB] \&.\&.\&.]\fR
.fi
.if n \{\
.RE
.\}
.PP
start,
stop, and
report
indicate which operation to perform\&. You can perform the designated operation for a single server or multiple servers, depending on the
\fIGNR\fR
list that follows the option name\&. If there is no list,
\fBmysqld_multi\fR
performs the operation for all servers in the option file\&.
.PP
Each
\fIGNR\fR
value represents an option group number or range of group numbers\&. The value should be the number at the end of the group name in the option file\&. For example, the
\fIGNR\fR
for a group named
[mysqld17]
is
17\&. To specify a range of numbers, separate the first and last numbers by a dash\&. The
\fIGNR\fR
value
10\-13
represents groups
[mysqld10]
through
[mysqld13]\&. Multiple groups or group ranges can be specified on the command line, separated by commas\&. There must be no whitespace characters (spaces or tabs) in the
\fIGNR\fR
list; anything after a whitespace character is ignored\&.
.PP
This command starts a single server using option group
[mysqld17]:
.sp
.if n \{\
.RS 4
.\}
.nf
shell> \fBmysqld_multi start 17\fR
.fi
.if n \{\
.RE
.\}
.PP
This command stops several servers, using option groups
[mysqld8]
and
[mysqld10]
through
[mysqld13]:
.sp
.if n \{\
.RS 4
.\}
.nf
shell> \fBmysqld_multi stop 8,10\-13\fR
.fi
.if n \{\
.RE
.\}
.PP
For an example of how you might set up an option file, use this command:
.sp
.if n \{\
.RS 4
.\}
.nf
shell> \fBmysqld_multi \-\-example\fR
.fi
.if n \{\
.RE
.\}
.PP
\fBmysqld_multi\fR
searches for option files as follows:
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: no-defaults option
.\" no-defaults option: mysqld_multi
With
\fB\-\-no\-defaults\fR, no option files are read\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: defaults-file option
.\" defaults-file option: mysqld_multi
With
\fB\-\-defaults\-file=\fR\fB\fIfile_name\fR\fR, only the named file is read\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: defaults-extra-file option
.\" defaults-extra-file option: mysqld_multi
Otherwise, option files in the standard list of locations are read, including any file named by the
\fB\-\-defaults\-extra\-file=\fR\fB\fIfile_name\fR\fR
option, if one is given\&. (If the option is given multiple times, the last value is used\&.)
.RE
.PP
Option files read are searched for
[mysqld_multi]
and
[mysqld\fIN\fR]
option groups\&. The
[mysqld_multi]
group can be used for options to
\fBmysqld_multi\fR
itself\&.
[mysqld\fIN\fR]
groups can be used for options passed to specific
\fBmysqld\fR
instances\&.
.PP
The
[mysqld]
or
[mysqld_safe]
groups can be used for common options read by all instances of
\fBmysqld\fR
or
\fBmysqld_safe\fR\&. You can specify a
\fB\-\-defaults\-file=\fR\fB\fIfile_name\fR\fR
option to use a different configuration file for that instance, in which case the
[mysqld]
or
[mysqld_safe]
groups from that file will be used for that instance\&.
.PP
\fBmysqld_multi\fR
supports the following options\&.
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: help option
.\" help option: mysqld_multi
\fB\-\-help\fR
.sp
Display a help message and exit\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: example option
.\" example option: mysqld_multi
\fB\-\-example\fR
.sp
Display a sample option file\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: log option
.\" log option: mysqld_multi
\fB\-\-log=\fR\fB\fIfile_name\fR\fR
.sp
Specify the name of the log file\&. If the file exists, log output is appended to it\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: mysqladmin option
.\" mysqladmin option: mysqld_multi
\fB\-\-mysqladmin=\fR\fB\fIprog_name\fR\fR
.sp
The
\fBmysqladmin\fR
binary to be used to stop servers\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: mysqld option
.\" mysqld option: mysqld_multi
\fB\-\-mysqld=\fR\fB\fIprog_name\fR\fR
.sp
The
\fBmysqld\fR
binary to be used\&. Note that you can specify
\fBmysqld_safe\fR
as the value for this option also\&. If you use
\fBmysqld_safe\fR
to start the server, you can include the
mysqld
or
ledir
options in the corresponding
[mysqld\fIN\fR]
option group\&. These options indicate the name of the server that
\fBmysqld_safe\fR
should start and the path name of the directory where the server is located\&. (See the descriptions for these options in
\fBmysqld_safe\fR(1)\&.) Example:
.sp
.if n \{\
.RS 4
.\}
.nf
[mysqld38]
mysqld = mysqld\-debug
ledir  = /opt/local/mysql/libexec
.fi
.if n \{\
.RE
.\}
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: no-log option
.\" no-log option: mysqld_multi
\fB\-\-no\-log\fR
.sp
Print log information to
stdout
rather than to the log file\&. By default, output goes to the log file\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: password option
.\" password option: mysqld_multi
\fB\-\-password=\fR\fB\fIpassword\fR\fR
.sp
The password of the MariaDB account to use when invoking
\fBmysqladmin\fR\&. Note that the password value is not optional for this option, unlike for other MariaDB programs\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: silent option
.\" silent option: mysqld_multi
\fB\-\-silent\fR
.sp
Silent mode; disable warnings\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: tcp-ip option
.\" tcp-ip option: mysqld_multi
\fB\-\-tcp\-ip\fR
.sp
Connect to the MariaDB server(s) via the TCP/IP port instead of the UNIX socket\&. This affects stopping
and reporting. If a socket file is missing, the server may still be running, but can be accessed only
via the TCP/IP port\&. By default connecting is done via the UNIX socket\&. This option affects
stop and report operations\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: user option
.\" user option: mysqld_multi
\fB\-\-user=\fR\fB\fIuser_name\fR\fR
.sp
The user name of the MariaDB account to use when invoking
\fBmysqladmin\fR\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: verbose option
.\" verbose option: mysqld_multi
\fB\-\-verbose\fR
.sp
Be more verbose\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: version option
.\" version option: mysqld_multi
\fB\-\-version\fR
.sp
Display version information and exit\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
.\" mysqld_multi: wsrep-new-cluster option
.\" wsrep-new-cluster option: mysqld_multi
\fB\-\-wsrep\-new\-cluster\fR
.sp
Bootstrap a cluster\&.
.RE
.PP
Some notes about
\fBmysqld_multi\fR:
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fBMost important\fR: Before using
\fBmysqld_multi\fR
be sure that you understand the meanings of the options that are passed to the
\fBmysqld\fR
servers and
\fIwhy\fR
you would want to have separate
\fBmysqld\fR
processes\&. Beware of the dangers of using multiple
\fBmysqld\fR
servers with the same data directory\&. Use separate data directories, unless you
\fIknow\fR
what you are doing\&. Starting multiple servers with the same data directory does
\fInot\fR
give you extra performance in a threaded system\&.
.sp .5v
.RE
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fBImportant\fR: Make sure that the data directory for each server is fully accessible to the Unix account that the specific
\fBmysqld\fR
process is started as\&.
\fIDo not\fR
use the Unix
\fIroot\fR
account for this, unless you
\fIknow\fR
what you are doing\&.
.sp .5v
.RE
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
Make sure that the MariaDB account used for stopping the
\fBmysqld\fR
servers (with the
\fBmysqladmin\fR
program) has the same user name and password for each server\&. Also, make sure that the account has the
SHUTDOWN
privilege\&. If the servers that you want to manage have different user names or passwords for the administrative accounts, you might want to create an account on each server that has the same user name and password\&. For example, you might set up a common
multi_admin
account by executing the following commands for each server:
.sp
.if n \{\
.RS 4
.\}
.nf
shell> \fBmysql \-u root \-S /tmp/mysql\&.sock \-p\fR
Enter password:
mysql> \fBGRANT SHUTDOWN ON *\&.*\fR
    \-> \fBTO 'multi_admin'@'localhost' IDENTIFIED BY 'multipass';\fR
.fi
.if n \{\
.RE
.\}
.sp
Change the connection parameters appropriately when connecting to each one\&. Note that the host name part of the account name must allow you to connect as
multi_admin
from the host where you want to run
\fBmysqld_multi\fR\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
The Unix socket file and the TCP/IP port number must be different for every
\fBmysqld\fR\&. (Alternatively, if the host has multiple network addresses, you can use
\fB\-\-bind\-address\fR
to cause different servers to listen to different interfaces\&.)
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
The
\fB\-\-pid\-file\fR
option is very important if you are using
\fBmysqld_safe\fR
to start
\fBmysqld\fR
(for example,
\fB\-\-mysqld=mysqld_safe\fR) Every
\fBmysqld\fR
should have its own process ID file\&. The advantage of using
\fBmysqld_safe\fR
instead of
\fBmysqld\fR
is that
\fBmysqld_safe\fR
monitors its
\fBmysqld\fR
process and restarts it if the process terminates due to a signal sent using
kill \-9
or for other reasons, such as a segmentation fault\&. Please note that the
\fBmysqld_safe\fR
script might require that you start it from a certain place\&. This means that you might have to change location to a certain directory before running
\fBmysqld_multi\fR\&. If you have problems starting, please see the
\fBmysqld_safe\fR
script\&. Check especially the lines:
.sp
.if n \{\
.RS 4
.\}
.nf
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
MY_PWD=`pwd`
# Check if we are starting this relative (for the binary release)
if test \-d $MY_PWD/data/mysql \-a \e
   \-f \&./share/mysql/english/errmsg\&.sys \-a \e
   \-x \&./bin/mysqld
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
.fi
.if n \{\
.RE
.\}
.sp
The test performed by these lines should be successful, or you might encounter problems\&. See
\fBmysqld_safe\fR(1)\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
You might want to use the
\fB\-\-user\fR
option for
\fBmysqld\fR, but to do this you need to run the
\fBmysqld_multi\fR
script as the Unix
root
user\&. Having the option in the option file doesn't matter; you just get a warning if you are not the superuser and the
\fBmysqld\fR
processes are started under your own Unix account\&.
.RE
.PP
The following example shows how you might set up an option file for use with
\fBmysqld_multi\fR\&. The order in which the
\fBmysqld\fR
programs are started or stopped depends on the order in which they appear in the option file\&. Group numbers need not form an unbroken sequence\&. The first and fifth
[mysqld\fIN\fR]
groups were intentionally omitted from the example to illustrate that you can have
\(lqgaps\(rq
in the option file\&. This gives you more flexibility\&.
.sp
.if n \{\
.RS 4
.\}
.nf
# This file should probably be in your home dir (~/\&.my\&.cnf)
# or /etc/my\&.cnf
# Version 2\&.1 by Jani Tolonen
[mysqld_multi]
mysqld     = /usr/local/bin/mysqld_safe
mysqladmin = /usr/local/bin/mysqladmin
user       = multi_admin
password   = multipass
[mysqld2]
socket     = /tmp/mysql\&.sock2
port       = 3307
pid\-file   = /usr/local/mysql/var2/hostname\&.pid2
datadir    = /usr/local/mysql/var2
language   = /usr/local/share/mysql/english
user       = john
[mysqld3]
socket     = /tmp/mysql\&.sock3
port       = 3308
pid\-file   = /usr/local/mysql/var3/hostname\&.pid3
datadir    = /usr/local/mysql/var3
language   = /usr/local/share/mysql/swedish
user       = monty
[mysqld4]
socket     = /tmp/mysql\&.sock4
port       = 3309
pid\-file   = /usr/local/mysql/var4/hostname\&.pid4
datadir    = /usr/local/mysql/var4
language   = /usr/local/share/mysql/estonia
user       = tonu
[mysqld6]
socket     = /tmp/mysql\&.sock6
port       = 3311
pid\-file   = /usr/local/mysql/var6/hostname\&.pid6
datadir    = /usr/local/mysql/var6
language   = /usr/local/share/mysql/japanese
user       = jani
.fi
.if n \{\
.RE
.\}
.PP
.SH "COPYRIGHT"
.br
.PP
Copyright 2007-2008 MySQL AB, 2008-2010 Sun Microsystems, Inc., 2010-2020 MariaDB Foundation
.PP
This documentation is free software; you can redistribute it and/or modify it only under the terms of the GNU General Public License as published by the Free Software Foundation; version 2 of the License.
.PP
This documentation is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
.PP
You should have received a copy of the GNU General Public License along with the program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335 USA or see http://www.gnu.org/licenses/.
.sp
.SH "SEE ALSO"
For more information, please refer to the MariaDB Documentation, available online at https://mariadb.com/docs/
.SH AUTHOR
MariaDB Foundation (http://www.mariadb.org/).


pcre/NEWS
News about PCRE releases
------------------------

Note that this library (now called PCRE1) is now being maintained for bug fixes
only. New projects are advised to use the new PCRE2 libraries.


Release 8.43 23-February-2019
-----------------------------

This is a bug-fix release.


Release 8.42 20-March-2018
--------------------------

This is a bug-fix release.


Release 8.41 13-June-2017
-------------------------

This is a bug-fix release.


Release 8.40 11-January-2017
----------------------------

This is a bug-fix release.


Release 8.39 14-June-2016
-------------------------

Some appropriate PCRE2 JIT improvements have been retro-fitted to PCRE1. Apart
from that, this is another bug-fix release. Note that this library (now called
PCRE1) is now being maintained for bug fixes only. New projects are advised to
use the new PCRE2 libraries.


Release 8.38 23-November-2015
-----------------------------

This is bug-fix release. Note that this library (now called PCRE1) is now being
maintained for bug fixes only. New projects are advised to use the new PCRE2
libraries.


Release 8.37 28-April-2015
--------------------------

This is bug-fix release. Note that this library (now called PCRE1) is now being
maintained for bug fixes only. New projects are advised to use the new PCRE2
libraries.


Release 8.36 26-September-2014
------------------------------

This is primarily a bug-fix release. However, in addition, the Unicode data
tables have been updated to Unicode 7.0.0.


Release 8.35 04-April-2014
--------------------------

There have been performance improvements for classes containing non-ASCII
characters and the "auto-possessification" feature has been extended. Other
minor improvements have been implemented and bugs fixed. There is a new callout
feature to enable applications to do detailed stack checks at compile time, to
avoid running out of stack for deeply nested parentheses. The JIT compiler has
been extended with experimental support for ARM-64, MIPS-64, and PPC-LE.


Release 8.34 15-December-2013
-----------------------------

As well as fixing the inevitable bugs, performance has been improved by
refactoring and extending the amount of "auto-possessification" that PCRE does.
Other notable changes:

.  Implemented PCRE_INFO_MATCH_EMPTY, which yields 1 if the pattern can match
   an empty string. If it can, pcretest shows this in its information output.

.  A back reference to a named subpattern when there is more than one of the
   same name now checks them in the order in which they appear in the pattern.
   The first one that is set is used for the reference. Previously only the
   first one was inspected. This change makes PCRE more compatible with Perl.

.  Unicode character properties were updated from Unicode 6.3.0.

.  The character VT has been added to the set of characters that match \s and
   are generally treated as white space, following this same change in Perl
   5.18. There is now no difference between "Perl space" and "POSIX space".

.  Perl has changed its handling of \8 and \9. If there is no previously
   encountered capturing group of those numbers, they are treated as the
   literal characters 8 and 9 instead of a binary zero followed by the
   literals. PCRE now does the same.

.  Following Perl, added \o{} to specify codepoints in octal, making it
   possible to specify values greater than 0777 and also making them
   unambiguous.

.  In UCP mode, \s was not matching two of the characters that Perl matches,
   namely NEL (U+0085) and MONGOLIAN VOWEL SEPARATOR (U+180E), though they
   were matched by \h.

.  Add JIT support for the 64 bit TileGX architecture.

.  Upgraded the handling of the POSIX classes [:graph:], [:print:], and
   [:punct:] when PCRE_UCP is set so as to include the same characters as Perl
   does in Unicode mode.

.  Perl no longer allows group names to start with digits, so I have made this
   change also in PCRE.

.  Added support for [[:<:]] and [[:>:]] as used in the BSD POSIX library to
   mean "start of word" and "end of word", respectively, as a transition aid.


Release 8.33 28-May-2013
--------------------------

A number of bugs are fixed, and some performance improvements have been made.
There are also some new features, of which these are the most important:

.  The behaviour of the backtracking verbs has been rationalized and
   documented in more detail.

.  JIT now supports callouts and all of the backtracking verbs.

.  Unicode validation has been updated in the light of Unicode Corrigendum #9,
   which points out that "non characters" are not "characters that may not
   appear in Unicode strings" but rather "characters that are reserved for
   internal use and have only local meaning".

.  (*LIMIT_MATCH=d) and (*LIMIT_RECURSION=d) have been added so that the
   creator of a pattern can specify lower (but not higher) limits for the
   matching process.

.  The PCRE_NEVER_UTF option is available to prevent pattern-writers from using
   the (*UTF) feature, as this could be a security issue.


Release 8.32 30-November-2012
-----------------------------

This release fixes a number of bugs, but also has some new features. These are
the highlights:

.  There is now support for 32-bit character strings and UTF-32. Like the
   16-bit support, this is done by compiling a separate 32-bit library.

.  \X now matches a Unicode extended grapheme cluster.

.  Case-independent matching of Unicode characters that have more than one
   "other case" now makes all three (or more) characters equivalent. This
   applies, for example, to Greek Sigma, which has two lowercase versions.

.  Unicode character properties are updated to Unicode 6.2.0.

.  The EBCDIC support, which had decayed, has had a spring clean.

.  A number of JIT optimizations have been added, which give faster JIT
   execution speed. In addition, a new direct interface to JIT execution is

support-files/mysqld_multi.server.sh
#!/bin/sh
#
# A simple startup script for mysqld_multi by Tim Smith and Jani Tolonen.
# This script assumes that my.cnf file exists either in /etc/my.cnf or
# /root/.my.cnf and has groups [mysqld_multi] and [mysqldN]. See the
# mysqld_multi documentation for detailed instructions.
#
# This script can be used as /etc/init.d/mysql.server
#
# Comments to support chkconfig on RedHat Linux
# chkconfig: 2345 64 36
# description: A very fast and reliable SQL database engine.
#
# Version 1.0
#

### BEGIN INIT INFO
# Provides:          mysqld_multi
# Required-Start: $local_fs $network $remote_fs
# Should-Start: ypbind nscd ldap ntpd xntpd
# Required-Stop: $local_fs $network $remote_fs
# Default-Start:  2 3 4 5
# Default-Stop: 0 1 6
# Short-Description: Start and stop multiple mysql database server daemon instances
# Description:       Controls multiple MariaDB database server daemon instances
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
NAME=mysqld_multi
DESC=mysqld_multi

basedir=/usr
bindir=/usr/bin

if test -x $bindir/mysqld_multi
then
  mysqld_multi="$bindir/mysqld_multi";
else
  echo "Can't execute $bindir/mysqld_multi from dir $basedir";
  exit;
fi

case "$1" in
    'start' )
        "$mysqld_multi" start $2
        ;;
    'stop' )
        "$mysqld_multi" stop $2
        ;;
    'report' )
        "$mysqld_multi" report $2
        ;;
    'restart' )
        "$mysqld_multi" stop $2
        "$mysqld_multi" start $2
        ;;
    *)
        echo "Usage: $0 {start|stop|report|restart}" >&2
        ;;
esac
